src/cljs/main/broadfcui/page/method_repo/methods_configs_acl.cljs

```diff
@@ -22,77 +23,79 @@
-            (:acl-vec @state))
-           [comps/Button {:text "Add new" :icon :add-new
-                          :onClick #(swap! state assoc :acl-vec
-                                           (conj (react/call :capture-ui-state this)
-                                                 {:user "" :role reader-level}))}]
-           [:label {:style {:cursor "pointer"}}
-            [:input {:type "checkbox" :ref "publicbox"
-                     :style {:marginLeft "2em" :verticalAlign "middle"}
-                     :onChange #(swap! state assoc :public-status (-> (@refs "publicbox") .-checked))
-                     :checked (:public-status @state)}]
-            [:span {:style {:paddingLeft 6 :verticalAlign "middle"}} "Publicly Readable?"]]
-           (style/create-validation-error-message (:validation-error @state))
-           [comps/ErrorViewer {:error (:save-error @state)}]]
-          (:error @state) (style/create-server-error-message (cond (= (:error @state) "Forbidden") (str "You are unauthorized to edit this " (clojure.string/lower-case (:entityType props)) ".")
-                                                                   :else (:error @state)))
-          :else [comps/Spinner {:text
-                                (str "Loading Permissions for " (:title props) "...")}]))
-       :ok-button (when (:acl-vec @state) {:text "Save" :onClick #(react/call :persist-acl this)})}])
+        (net/render-with-ajax
+         (:acl-response @state)
+         #(this :-render-acl-form)
+         {:loading-text (str "Loading Permissions for " (:title props) "...")
+          :error-override
+          #(net/overwrite-error
+            (str "You are unauthorized to edit this "
+                 (clojure.string/lower-case (:entityType props)) ".")
+            403 %)}))
+       :ok-button (when (:acl-vec @state) {:text "Save" :onClick #(this :-persist-acl)})}])
    :component-did-mount
    (fn [{:keys [props state]}]
      (endpoints/call-ajax-orch
       {:endpoint (:load-endpoint props)
-       :on-done (fn [{:keys [success? get-parsed-response status-text]}]
-                  (if success?
-                    (let [response-vec (mapv utils/keywordize-keys (get-parsed-response false))
-                          acl-vec (filterv #(not= "public" (:user %)) response-vec)
-                          public-user (first (filter #(= "public" (:user %)) response-vec))
-                          public-status (or (:role public-user) no-access-level)]
-                      (swap! state assoc :acl-vec acl-vec
-                             :public-status (= public-status reader-level)
-                             :count-orig (count acl-vec)))
-                    (swap! state assoc :error status-text)))}))
-   :persist-acl
+       :on-done (net/handle-ajax-response
+                 (fn [k v]
+                   (swap! state assoc-in [:acl-response k] v)
+                   (let [acl-vec (filterv #(not= "public" (:user %)) v)
```

Comment:
Instead of saving each of these into state as well, consider moving this let into the render method.